tenant/service_user.go

```diff
@@ -247,6 +276,49 @@ func encryptPassword(password string) (string, error) {
 	return string(passHash), nil
 }
 
+var classes []func(rune) bool = []func(rune) bool{
+	unicode.IsNumber,
+	unicode.IsUpper,
+	unicode.IsLower,
+	func(r rune) bool {
+		found := false
+		for _, s := range errors.SpecialChars {
+			found = (s == r) || found
+		}
+		return found
+	},
+}
+
+// IsPasswordStrong checks if a password is strong enough.
+func IsPasswordStrong(password string, doCheck bool) error {
+	const numClassesRequired = 3
+	var eSlice []error = nil
+	l := len(password)
+	if l < errors.MinPasswordLen || l > errors.MaxPasswordLen {
+		eSlice = append(eSlice, errors.EPasswordLength)
+	}
+	if doCheck {
+		// make a password copy that is the length of the max password length
+		constLenPassword := strings.Repeat(password, 1+(errors.MaxPasswordLen/len(password)))[:errors.MaxPasswordLen]
+		n := 0
+
+		// Walk the whole string for each class, for constant time operation
+		for _, f := range classes {
+			found := false
+			for _, r := range constLenPassword {
+				found = f(r) || found
+			}
+			if found {
+				n++
+			}
+		}
+		if n < numClassesRequired {
+			eSlice = append(eSlice, errors.EPasswordChars)
+		}
+	}
+	return eBase.Join(eSlice...)
```

Comment:
Let's join all the `tSlice`s too:
```
eBase.Join(tSlice...) // for constant time

return eBase.Join(eSlice...)
```

Perhaps we also want to return the result of `eBase.Join(tSlice...)` so the consumer can do the same thing with it as it does with `eBase.Join(eSlice...)` (without displaying to the user). I feel we're going down a rabbit hole...